src/APICore.ts

```diff
@@ -2,7 +2,10 @@ import {APIConfiguration} from './ConfigurationInterfaces';
 import {ResponseHandler} from './handlers/ResponseHandlerInterfaces';
 import handleResponse, {defaultResponseHandlers} from './handlers/ResponseHandlers';
 
-export default class API {
+type APIPrototype = typeof API.prototype;
+export type IAPI = {[P in keyof APIPrototype]: APIPrototype[P]};
```

Comment:
What is the goal of all this dark sorcery ?

Reply:
Generating the interface from the class instead of writing an interface implemented by the class

Reply:
Isn't it a circular dependency? Like that the API class implements the IAPI interface with is derived from the API class prototype. My mind is blown

Reply:
Yeah, I tried it and it somehow worked 🤷‍♂